### ISSUE TITLE
Array.push is broken

### ISSUE THREAD

**Issue by cdaniel** [Automattic/mongoose#5619]
Up until version 4.11.9 I had no issues with Array.push. However, I cannot update to 4.11.10, because I get following error:

```
TypeError: Model is not a constructor
    at Array._cast (/home/kdaniel/git/atlas2/node_modules/mongoose/lib/types/array.js:120:17)
    at Array._mapCast (/home/kdaniel/git/atlas2/node_modules/mongoose/lib/types/array.js:296:17)
    at Object.map (native)
    at Array.push (/home/kdaniel/git/atlas2/node_modules/mongoose/lib/types/array.js:328:25)
    at model._MapSchema.methods.formASubmap (/home/kdaniel/git/atlas2/src-server/workspace/model/map-schema.js:487:21)
    at /home/kdaniel/git/atlas2/src-server/workspace/workspace-router.js:410:26
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```
The code is relatively simple and worked before:
```
        var submapNode = new Node({
            _id: submapNodeID,
            otherproperties: ...   
        });
        _this.nodes.push(submapNode);
```
The field `nodes` is populated. If I do not populate it, the push will work fine, however, I need it to be populated. Possibly, this is related to changes introduced here https://github.com/Automattic/mongoose/commit/60331797ac20cac3f615b0b23aa785285efa1750.

**Comment by vkarpov15** [comment]
We fixed a related issue here: https://github.com/Automattic/mongoose/commit/a227d6f70d66fc2003461a857bb2cc16caa809b4 so this might be fixed in master. However, I'm unable to repro your issue, the below script works fine for me in 4.11.10. Can you please modify the below script to reproduce your issue?

```javascript
const mongoose = require('mongoose');

var nodeSchema = new mongoose.Schema({});
var parentSchema = new mongoose.Schema({
  nodes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Node' }]
});

var Node = mongoose.model('Node', nodeSchema);
var Parent = mongoose.model('Parent', parentSchema);

run().catch(error => console.error(error));

async function run() {
  await mongoose.connect('mongodb://localhost:27017/test');

  let doc = await Parent.create({});
  doc = await Parent.findById(doc._id);
  doc = await doc.populate('nodes').execPopulate();
  doc.nodes.push(new Node());
  console.log('done');
}
```

**Comment by drupdateio** [comment]
Same issue for me

**Comment by guillaumev** [comment]
Same issue for me

**Comment by cdaniel** [comment]
This code works fine with 4.11.9 but not with 4.11.10:
```
var mongoose = require('mongoose');
var Q = require('q');

var connection = mongoose.connect('mongodb://localhost:27017/test');

var nodeSchema = new mongoose.Schema({});

var parentSchema = new mongoose.Schema({nodes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Node' }]});

var Node = mongoose.model('Node', nodeSchema);

parentSchema.methods.showtime = function() {
  var promises = [];
  var node = new Node({});
  var _this = this;
  _this.nodes.push(node);

  promises.push(node.save());
  promises.push(_this.save());

  return Q.allSettled(promises).then(function(){
    return _this;
  });
};


var Parent = mongoose.model('Parent', parentSchema);


try {
    Q.fcall(function(){
      return new Parent({}).save();
    })
    .then(function(parent){
      return parent.showtime();
    })
    .then(function(parent){
      return Parent.findById(parent._id);
    })
    .then(function(parentReloaded){
      return parentReloaded.populate('nodes').execPopulate();
    })
    .then(function(parentReloaded){
      return parentReloaded.showtime();
    })
    .done(function(parentReloaded,error){
      mongoose.connection.close();
      console.log(' > END <');
    });


} catch (e){
  console.log(e);
}

```

**Comment by vkarpov15** [comment]
@cdaniel yep this issue was fixed with 4.11.11, please upgrade and it should go away. Thanks for the repro script.
